packages/@aws-cdk/aws-lambda-golang/README.md

```diff
@@ -0,0 +1,246 @@
+# Amazon Lambda Golang Library
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+> The APIs of higher level constructs in this module are experimental and under active development.
+> They are subject to non-backward compatible changes or removal in any future version. These are
+> not subject to the [Semantic Versioning](https://semver.org/) model and breaking changes will be
+> announced in the release notes. This means that while you may use them, you may need to update
+> your source code when upgrading to a newer version of this package.
+
+---
+
+<!--END STABILITY BANNER-->
+
+This library provides constructs for Golang Lambda functions.
+
+To use this module you will need to have Docker installed.
+
+This module also requires that your Golang application is
+using a Go version >= 1.11 and is using [Go modules](https://golang.org/ref/mod).
+
+## Golang Function
+
+Define a `GolangFunction`:
+
+```ts
+new lambda.GolangFunction(this, 'handler', {
+  entry: 'app/cmd/api'
+});
+```
+
+By default, if a folder path is provided as the `entry`, the construct will assume there is a go entry file (i.e. `main.go`).
+Let's look at an example Go project:
+
+```bash
+lamda-app
+├── cmd
+│   └── api
+│       └── main.go
+├── go.mod
+├── go.sum
+├── pkg
+│   ├── auth
+│   │   └── auth.go
+│   └── middleware
+│       └── middleware.go
+└── vendor
+    ├── github.com
+    │   └── aws
+    │       └── aws-lambda-go
+    └── modules.txt
+```
+
+With the above layout I could either provide the `entry` as `lambda-app/cmd/api` or `lambda-app/cmd/api/main.go`, either will work.
+When the construct builds the golang binary this will be translated `go build ./cmd/api` & `go build ./cmd/api/main.go` respectively.
+The construct will figure out where it needs to run the `go build` command from, in this example it would be from
+the `lambda-app` directory. It does this by determining the [mod file path](#mod-file-path), which is explained in the
+next section.
+
+### mod file path
+
+The `GolangFunction` tries to automatically determine your project root, that is
+the root of your golang project. This is usually where the top level `go.mod` file or
+`vendor` folder of your project is located. When bundling in a Docker container, the
+`modFilePath` is used as the source (`/asset-input`) for the volume mounted in
+the container.
+
+The CDK will walk up parent folders starting from
+the current working directory until it finds a folder containing a `go.mod` file.
+
+Alternatively, you can specify the `modFilePath` prop manually. In this case you
```

Comment:
It's not clear if `modFilePath` is the path to the `go.mod` file or the directory in which the `go.mod` file exists.

I am wondering if perhaps it would be more straightforward to call this `moduleDir`